src/InteractiveWindow/Editor/InteractiveWindow_UIThread.cs

```diff
@@ -214,12 +213,13 @@ public Task<ExecutionResult> ResetAsync(bool initialize)
                     _window._currentLanguageBuffer = null;
                 }
 
-                // replace the task being interrupted by a "reset" task:
                 State = State.Resetting;
-                _currentTask = _window._evaluator.ResetAsync(initialize);
-                _currentTask.ContinueWith(FinishExecute, _uiScheduler);
+                var executionResult = await _window._evaluator.ResetAsync(initialize).ConfigureAwait(true);
```

Comment:
Note this is a potential change of behavior: previously this would schedule FinishExecute to the UI thread, even if ResetAsync was called on the background thread. Now it will stay on the background thread. You'll probably want to add an assert to ensure this is running where you think it is now.